src/java/org/apache/cassandra/nodes/virtual/NodeConstants.java

```diff
@@ -38,15 +38,21 @@ public class NodeConstants
     public static final String PEERS_V2_VIEW_NAME = "peer_v2_nodes";
     public static final String LEGACY_PEERS_VIEW_NAME = "peer_nodes";
 
-    public static final Set<String> ALL_TABLES = Sets.newHashSet(LOCAL, PEERS_V2, LEGACY_PEERS);
-    public static final Set<String> ALL_VIEWS = Sets.newHashSet(LOCAL_VIEW_NAME, PEERS_V2_VIEW_NAME, LEGACY_PEERS_VIEW_NAME);
+    public static final Set<String> ALL_TABLES = ImmutableSet.of(LOCAL, PEERS_V2, LEGACY_PEERS);
+    public static final Set<String> ALL_VIEWS = ImmutableSet.of(LOCAL_VIEW_NAME, PEERS_V2_VIEW_NAME, LEGACY_PEERS_VIEW_NAME);
 
     public static boolean canBeMapped(String keyspace, String table)
     {
         if (keyspace.equals(SchemaConstants.SYSTEM_VIEWS_KEYSPACE_NAME))
             return ALL_VIEWS.contains(table);
         if (keyspace.equals(SchemaConstants.SYSTEM_KEYSPACE_NAME))
+        {
+            if (table.equals(LOCAL) || table.equals(LEGACY_PEERS))
+            {
+                return !LegacySystemKeyspaceToNodes.isRunning();
+            }
             return ALL_TABLES.contains(table);
+        }
```

Comment:
trivial and optional
```suggestion
        if (keyspace.equals(SchemaConstants.SYSTEM_KEYSPACE_NAME))
            if (table.equals(LOCAL) || table.equals(LEGACY_PEERS))
                return !LegacySystemKeyspaceToNodes.isRunning();
            return ALL_TABLES.contains(table);
```

Reply:
Fixed.